### ISSUE TITLE
File Selector Demo

### ISSUE THREAD

**Issue by vedat42** [DelphiWorlds/Kastri#139]
Hi, I want to list all the .mp3 file with it's located path in a listbox. Then play the clicked mp3 file. I didn't understand properly if it is possible by using TFilesSelector. It would be awesome if I could get any help

**Comment by DelphiWorlds** [comment]
What will you be using to play the `.mp3` files?

**Comment by vedat42** [comment]
I will use TMediaPlayer

**Comment by DelphiWorlds** [comment]
In that case, you could use `TAndroidHelperEx.ImportFile` as described in [the readme](https://github.com/DelphiWorlds/Kastri/tree/master/Demos/FilesSelector#readme). For example, you could change the ListBox1ItemClick method in the demo to look like this:

```
procedure TForm1.ListBox1ItemClick(const Sender: TCustomListBox; const Item: TListBoxItem);
var
  LDisplayName, LFileName: string;
begin
  LDisplayName := FSelector.SelectedFiles[Item.Index].DisplayName;
  DisplayNameValueLabel.Text := LDisplayName;
  if LDisplayName.EndsWith('mp3', True) then
  begin
    LFileName := TPath.Combine(TPath.GetDocumentsPath, FSelector.SelectedFiles[AIndex].DisplayName);
    TAndroidHelperEx.ImportFile(FSelector.SelectedFiles[AIndex].RawPath, LFileName);
    MediaPlayer1.FileName := LFileName;
    MediaPlayer1.Play;
   end;
end;
```

**Comment by vedat42** [comment]
Thank you for reply. When I select an .mp3 file in the download folder the full path is shown like that : '/data/user/0/com.embarcadero.FilesSelectorDemo/files/Download..' . It doesn't show mp3 file's name but ends with **Download..** Hence, I am not able to play the file.

**Comment by DelphiWorlds** [comment]
Did you look at the code I provided in my reply? It uses the `DisplayName` property of the selected files to construct the filename

**Comment by vedat42** [comment]
Yes, I looked and applied the code that you showed. I used **DisplayName** property of the **SelectedFiles** but was not able to display the filename with full path. Let me write down the code scope that I added. Under the **Pics and Movies** button click event I added Audio file option as follow :
`FSelector.FileKinds := [TFileKind.Image, TFileKind.Movie, TFileKind.Audio];`
`FSelector.Select(TSelectionMode.Content);`

In the **SelectorCompleteHandler** event, I've added :
`var`
`LFileName,LDisplayName: string; `
`begin`
`ListBox1.Clear;`
`for I := 0 to Length(FSelector.SelectedFiles) - 1 do`
`begin`
    `LDisplayName := FSelector.SelectedFiles[I].DisplayName;`
    `DisplayNameValueLabel.Text := LDisplayName;`
    `LFileName := TPath.Combine(TPath.GetDocumentsPath, LDisplayName);`
    `TAndroidHelperEx.ImportFile(FSelector.SelectedFiles[I].RawPath, LFileName);`
    `ListBox1.Items.Add(LFileName);`
 `end;`
 `end;`

Here I just wanted to display the full path in the ListBox to see if it is shown correctly. But as I said, path wasn't displayed correctly.
